database/main.go

```diff
@@ -108,11 +120,38 @@ func saveWorkspaceDB(
 	}
 
 	if err := g.WriteFileInWorkspace(ctx, dbWorkspacePath, updatedDBData, gptscript.WriteFileInWorkspaceOptions{
-		WorkspaceID: workspaceID,
+		WorkspaceID:      workspaceID,
+		CreateRevision:   &([]bool{true}[0]),
+		LatestRevisionID: revisionID,
 	}); err != nil {
 		return fmt.Errorf("Error writing updated DB file to workspace: %v", err)
 	}
 
+	// Delete old revisions after successfully writing the new revision
+	revisions, err := g.ListRevisionsForFileInWorkspace(ctx, dbWorkspacePath, gptscript.ListRevisionsForFileInWorkspaceOptions{
+		WorkspaceID: workspaceID,
+	})
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Error listing revisions: %v\n", err)
+		return nil
+	}
+
+	lastRevisionIndex := slices.IndexFunc(revisions, func(rev gptscript.FileInfo) bool {
+		return rev.RevisionID == revisionID
+	})
+
+	if lastRevisionIndex < 0 {
+		return nil
+	}
+
+	for _, rev := range revisions[:lastRevisionIndex+1] {
```

Comment:
This seems to be delete ALL revisions. Is that the intention?

Reply:
Iiuc, this drops all revisions up-to and including the previous revision (not the one we just created).